core/src/main/resources/org/apache/spark/ui/static/executorspage.js

```diff
@@ -408,12 +420,17 @@ $(document).ready(function () {
                             data: 'id', render: function (data, type) {
                                 return type === 'display' ? ("<a href='threadDump/?executorId=" + data + "'>Thread Dump</a>" ) : data;
                             }
-                        }
+                        },
+                        {data: 'worker', render: formatWorkersCells}
                     ],
                     "columnDefs": [
                         {
                             "targets": [ 16 ],
                             "visible": getThreadDumpEnabled()
+                        },
+                        {
+                            "targets": [ 17 ],
+                            "visible": workersExist(response)
```

Comment:
We ran into an issue recently with columnDefs, it only runs on initial page load and doesn't catch changes on refresh. So for columns with non-static visibility you should add a line like this one below `dt.column(15).visible(logsExist(response));`